parlai/core/metrics.py

```diff
@@ -247,23 +250,22 @@ def report(self):
         m['total'] = total
         if total > 0:
             if self.print_prediction_metrics or self.metrics['f1'] > 0:
```

Comment:
why is f1 > 0 in there?

Reply:
so the self.print_prediction_metrics is only there to make it so that if f1 == 0 because the model got everything completely wrong, then it will still print.

f1 > 0 is there because that flag is not shared during hogwild, and we definitely always want to enter this section when f1 > 0. We unfortunately aren't entering when f1 is zero but the model is trying, but that fix would be more complex and I didn't want @ ekmb to do that for now.

Reply:
i dont understand sorry. metrics has got too complicated lol.
under what circumstances do we not want to print?

Reply:
i don't understand, but hopefully you know what you are doing :)
why does this has to depend on f1, and not some other metric?